File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        boolean mode = false;
        String ret = "";
        for(int i = 0; i < code.length(); i++){
            char c = code.charAt(i);
            if(!mode){
                if(c != '1' && i % 2 == 0){
                    ret += code.charAt(i);
                }
                else if(c == '1') mode = true;
            }
            else{
                if(c != '1' && i % 2 == 1){
                    ret += code.charAt(i);
                }
                else if(c == '1') mode = false;
            }
        }
        if(ret.isEmpty()) ret = "EMPTY";
        return ret;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
    let ret = '';
    let mode = 0;
    for(let i = 0; i < code.length; i++){
        if(code[i] == '1'){
            mode = !mode;
            continue;
        }
        if(i % 2 == mode){
            ret += code[i];
        }
    }
    return ret.length == 0 ? "EMPTY" : ret;
}